library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/SessionReplayLifecycleCallbackTest.kt

```diff
@@ -65,43 +81,63 @@ class SessionReplayLifecycleCallbackTest {
     }
 
     @Test
-    fun `M start recording activity W onActivityResumed()`() {
+    fun `M start register fragment lifecycle W onActivityCreated(){FragmentActivity}`() {
         // Given
-        val mockActivity: Activity = mock()
+        val mockFragmentManager: FragmentManager = mock()
+        val mockFragmentActivity: FragmentActivity = mock {
+            whenever(it.supportFragmentManager).thenReturn(mockFragmentManager)
+        }
 
+        // When
+        testedCallback.onActivityCreated(mockFragmentActivity, mock())
+
+        // Then
+        val argumentCaptor = argumentCaptor<FragmentLifecycleCallbacks>()
+        verify(mockFragmentManager).registerFragmentLifecycleCallbacks(
+            argumentCaptor.capture(),
+            eq(true)
+        )
+        assertThat(argumentCaptor.firstValue)
+            .isInstanceOf(RecorderFragmentLifecycleCallback::class.java)
+    }
+
+    @Test
+    fun `M do nothing W onActivityCreated(){no FragmentActivity}`() {
+        // When
+        testedCallback.onActivityCreated(mockActivity, mock())
+
+        // Then
+        verifyZeroInteractions(mockActivity)
+    }
+
+    @Test
+    fun `M start recording activity W onActivityResumed()`() {
         // When
         testedCallback.onActivityResumed(mockActivity)
 
         // Then
-        verify(mockRecoder).startRecording(mockActivity)
-        assertThat(testedCallback.resumedActivities).containsKey(mockActivity)
+        verify(mockRecoder).startRecording(eq(listOf(mockWindow)), eq(mockActivity))
```

Comment:
You are right, reason why I added those like that was cause I thought it will do the assertion by reference in cause of the lists. Apparently is doing it by value anyway without the `eq`